aries/aries_vcx_core/src/anoncreds/anoncreds.rs

```diff
@@ -0,0 +1,1799 @@
+                        .and_then(|req_preds| req_preds.get(&reft))
+                })
+                .ok_or(AriesVcxCoreError::from_msg(
+                    // should not happen
+                    AriesVcxCoreErrorKind::InvalidState,
+                    format!("Unknown referent: {}", reft),
+                ))?;
+
+            let name = requested_val.get("name");
+            let names = requested_val.get("names").and_then(|v| v.as_array());
+
+            let attr_names = match (name, names) {
+                (Some(name), None) => vec![_normalize_attr_name(name.try_as_str()?)],
+                (None, Some(names)) => names
+                    .iter()
+                    .map(|v| v.try_as_str().map(_normalize_attr_name))
+                    .collect::<Result<_, _>>()?,
+                _ => Err(AriesVcxCoreError::from_msg(
+                    AriesVcxCoreErrorKind::InvalidInput,
+                    "exactly one of 'name' or 'names' must be present",
+                ))?,
+            };
+
+            let non_revoked = requested_val.get("non_revoked"); // note that aca-py askar fetches from proof_req json
+            let restrictions = requested_val.get("restrictions");
+
+            let credx_creds = self
+                ._get_credentials_for_proof_req_for_attr_name(wallet, restrictions, attr_names)
+                .await?;
+
+            let mut credentials_json = vec![];
+
+            for (cred_id, credx_cred) in credx_creds {
+                credentials_json.push(json!({
+                    "cred_info": _make_cred_info(&cred_id, &credx_cred)?,
+                    "interval": non_revoked
+                }))
+            }
+
+            cred_by_attr[ATTRS][reft] = Value::Array(credentials_json);
+        }
+
+        Ok(serde_json::to_string(&cred_by_attr)?)
+    }
+
+    async fn prover_create_credential_req(
+        &self,
+        wallet: &impl BaseWallet,
+        prover_did: &str,
+        cred_offer_json: &str,
+        cred_def_json: &str,
+        master_secret_id: &str,
+    ) -> VcxCoreResult<(String, String)> {
+        let prover_did = Did::parse(prover_did.to_owned())
+            .map_err(|e| AriesVcxCoreError::from_msg(AriesVcxCoreErrorKind::InvalidDid, e))?;
+        let mut cred_def_json: Value = serde_json::from_str(cred_def_json)?;
+        let cred_def_id = cred_def_json["id"].as_str().unwrap().to_string();
+        cred_def_json.as_object_mut().unwrap().insert(
+            "issuerId".to_owned(),
+            cred_def_id.split(':').next().unwrap().into(),
+        );
+        let cred_def: CredentialDefinition = serde_json::from_str(&cred_def_json.to_string())?;
+        let credential_offer: CredentialOffer = serde_json::from_str(cred_offer_json)?;
+        let link_secret = self.get_link_secret(wallet, master_secret_id).await?;
+
+        let (cred_req, cred_req_metadata) = anoncreds::prover::create_credential_request(
+            None,
+            Some(prover_did.did()),
```

Comment:
Nitpick: it would be preferable to swap these. Provide first argument `entropy` of value `Some(prover_did)`, and provide `None` for the 2nd arg. 
I don't know why this became 2 arguments in anoncreds-rs, originally it was `prover_did` later renamed to `entropy` to reflect actual purpose of the parameter. But there was a shift from dealing with "prover_did" to more generally "some entropy".